eldoc-box.el

```diff
@@ -578,6 +578,62 @@ display the docs in echo area depending on
     (when (eldoc-box--eldoc-message-function "%s" doc)
       (eldoc-display-in-echo-area docs interactive))))
 
+(defun eldoc-box--visible-p (&optional frame)
+  "Return non-nil if FRAME is visible."
+  (let ((frame (or frame eldoc-box--frame)))
+    (and (frame-live-p frame) (frame-visible-p frame))))
+
+(defun eldoc-box-scroll-up (&optional n)
+  "Scroll text of eldoc-box window upward N lines."
+  (interactive "p")
+  (if (eldoc-box--visible-p)
+      (with-selected-frame eldoc-box--frame
+        (with-current-buffer eldoc-box--buffer
+          (scroll-up n)))
+    (scroll-other-window n)))
+
+(defun eldoc-box-scroll-down (&optional n)
+  "Scroll text of eldoc-box window down N lines."
+  (interactive "p")
+  (eldoc-box-scroll-up (- (or n 1))))
+
+(defun eldoc-box-end (&optional n)
+  "Scroll text of eldoc-box window to the end.
+
+With numeric arg N, put window N/10 of the way from the end."
+  (interactive "P")
+  (if (eldoc-box--visible-p)
+      (with-selected-frame eldoc-box--frame
+        (with-current-buffer eldoc-box--buffer
+          (with-no-warnings
+            (end-of-buffer n))))
+    (end-of-buffer-other-window n)))
+
+(defun eldoc-box-beginning (&optional n)
+  "Scroll text of eldoc-box window to the beginning.
+
+With numeric arg N, put window N/10 of the way from the beginning."
+  (interactive "P")
+  (eldoc-box-end (- 10 (if (numberp n) n 0))))
+
+(defvar eldoc-box-hover-mode-map
```

Comment:
Instead of setting the keymap directly, could you define a separate keymap, say eldoc-box-extra-commands-map or something, so that people who want the functionality could set eldoc-box-hover-mode-map to that map or inherit that map?

Reply:
eldoc-box doesn't have a default mode map, when I want to add customzied
behavior I had to write like this:

```emacs-lisp
(defvar-keymap eldoc-box-map
  :doc "Additional keymap activated in eldoc-box mode."
  "<remap> <scroll-other-window>" #'eldoc-box-scroll-up
  "<remap> <scroll-other-window-down>" #'eldoc-box-scroll-down
  "<remap> <beginning-of-buffer-other-window>" #'eldoc-box-beginning
  "<remap> <end-of-buffer-other-window>" #'eldoc-box-end)

(add-to-list 'minor-mode-map-alist
             `(eldoc-box-hover-mode . ,eldoc-box-map))
```

If I write it like this directly, Emacs will throw an error:

```emacs-lisp
(use-package eldoc-box
  :bind (:map eldoc-box-hover-mode-map
              ;; Some bindings.
              ))
```

So I added a default map here. And it doesn't hardcoded any keybinds.

I had very little knowledge about minor mode keymapping, can you point
out how I can change the code like you said?

Reply:
Sorry, I should've been more clear. I was hoping that you could add two empty minor mode map for `eldoc-box-hover-/at-point-mode`, like you did in the PR. Then addd a third map with your bindings. Then if someone wants to use the bindings you defined, they can set the parent of, say, `eldoc-box-hover-mode-map` to your map, with `set-keymap-parent`. 

Basically I just want the bindings to be opt-in. People won't be happy if they update eldoc-box and suddenly scroll-other-window scrolls the childframe instead.

Reply:
> Basically I just want the bindings to be opt-in. People won't be happy
> if they update eldoc-box and suddenly scroll-other-window scrolls the
> childframe instead.

I see. The reason I wrote the code like this is because I encountered
such situation during using:

I split two windows when writing code (I'll call it A and B: A is the
window I'm editing, B is the window left), when eldoc-box's childframe
popup (let's call it C), it will cover the unused window B.  Since C
popup, it means that my main focus is on the popup window C, I didn't
care the covered window B's content.  When I want to
`scroll-other-window` which actually means I want to scroll the popup
window C. If I want to scroll the covered window B, I should hide the
popup window C.  And that's why I add an `else` here:

https://github.com/casouri/eldoc-box/pull/73/files#diff-a84c5ae0ee463811883a3f9d1d9ce0623efbc560c7de465cf1c74a96c996c873R605

But after writing these, I realized that your thought is more
reasonable: it may still have an need to scroll window B while keeping
childframe C popup.